lms/djangoapps/certificates/tests/test_queue.py

```diff
@@ -73,6 +78,58 @@ def test_no_create_action_in_queue_for_html_view_certs(self):
         self.assertEqual(certificate.status, CertificateStatuses.downloadable)
         self.assertIsNotNone(certificate.verify_uuid)
 
+    @ddt.data(
+        ('honor', 'honor'),
+        ('audit', 'audit')
+    )
+    @ddt.unpack
+    def test_add_cert_with_honor_certificates(self, mode, expected_mode):
+        """Test certificates generations for honor and audit modes."""
+        template_name = 'certificate-template-{id.org}-{id.course}.pdf'.format(
+            id=self.course.id
+        )
+        self.assert_queue_response(mode, expected_mode, template_name)
+
+    @ddt.data(
+        ('credit', 'verified'),
+        ('verified', 'verified')
+    )
+    @ddt.unpack
+    def test_add_cert_with_verified_certificates(self, mode, expected_mode):
+        """Test if enrollment mode is verified or credit along with valid
+        software-secure verification than verified certificate should be generated.
+        """
+        template_name = 'certificate-template-{id.org}-{id.course}-verified.pdf'.format(
+            id=self.course.id
+        )
+
+        self.assert_queue_response(mode, expected_mode, template_name)
+
+    def assert_queue_response(self, mode, expected_mode, expected_template_name):
+        """Dry method for course enrollment and adding request to queue."""
+        CourseEnrollmentFactory(
+            user=self.user_2,
+            course_id=self.course.id,
+            is_active=True,
+            mode=mode,
+        )
+        with patch('courseware.grades.grade', Mock(return_value={'grade': 'Pass', 'percent': 0.75})):
+            with patch.object(XQueueInterface, 'send_to_queue') as mock_send:
+                mock_send.return_value = (0, None)
+                self.xqueue.add_cert(self.user_2, self.course.id)
+
+        # Verify that the task was sent to the queue with the correct callback URL
+        self.assertTrue(mock_send.called)
+        __, kwargs = mock_send.call_args_list[0]
+
+        actual_header = json.loads(kwargs['header'])
+        self.assertIn('https://edx.org/update_certificate?key=', actual_header['lms_callback_url'])
```

Comment:
Where is this URL coming from? We should avoid hardcoding edX-specific URLs when possible.